src/viam/sdk/module/service.cpp

```diff
@@ -215,31 +213,53 @@ ::grpc::Status ModuleService_::Ready(::grpc::ServerContext* context,
     return grpc::Status();
 };
 
-ModuleService_::ModuleService_(std::string addr) {
-    module_ = std::make_shared<Module>(addr);
+ModuleService::ModuleService(std::string addr)
+    : module_(std::make_shared<Module>(std::move(addr))),
+      server_(std::make_shared<Server>()),
+      signal_manager_(std::make_unique<SignalManager>()) {}
+
+ModuleService::ModuleService(int argc,
+                             char** argv,
+                             std::vector<std::shared_ptr<ModelRegistration>> registrations) {
+    if (argc < 2) {
+        throw std::runtime_error("Need socket path as command line argument");
+    }
+    module_ = std::make_shared<Module>(argv[1]);
+    server_ = std::make_shared<Server>();
+    signal_manager_ = std::make_unique<SignalManager>();
+    set_logger_severity_from_args(argc, argv);
+
+    for (const std::shared_ptr<ModelRegistration>& mr : registrations) {
```

Comment:
`for (const auto& mr :` or `for (auto&& mr :`

Reply:
Sounds good; used the first. Just curious, it seems like you prefer `auto` when type annotations are too long? Or is it when they're obvious?

Reply:
I'm fairly pro-`auto` in general. In this case the type is obvious. Some interesting reading here: https://herbsutter.com/2013/08/12/gotw-94-solution-aaa-style-almost-always-auto/.

Reply:
Ah interesting stuff, thanks for the reading material. My intuition was that `auto` was like "cheating" in C++ and it was usually better to declare type. That article was insightful.